### PR TITLE
LVRUBY-100: Sorting news by views

### PR DESCRIPTION
**Pull request by kmandziy**
## JIRA ticket

http://ssu-jira.softserveinc.com/secure/RapidBoard.jspa?rapidView=979&view=planning&selectedIssue=LVRUBY-100

## Code reviewers

- [ ] @nataliakozoriz 

## Summary of issue

Added ability to see the most popular news

## Summary of change

Added AJAX requests to news.js file
Created index.js.haml for rendering news partial with AJAX
Screenshot from 2019-05-16 17-51-36

## Review checklist

- [ ] Rubocop doesn't raise any warnings and offenses
- [ ] New features are covered with new specs
- [ ] All specs pass

### CODE REVIEW

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **97.44%** covered at **4.48** hits/line

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **97.16%** covered at **4.11** hits/line

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **97.16%** covered at **4.11** hits/line

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **97.16%** covered at **4.11** hits/line

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **96.2%** covered at **3.91** hits/line

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **96.2%** covered at **3.91** hits/line

File: app/models/news.rb [hunk 1]
```diff
@@ -10,10 +10,11 @@ class News < ApplicationRecord
   validates :body, presence: true
   validates :short_information, presence: true, length: { maximum: 200 }
 
+  scope :by_views, -> { order(views: :desc) }
   scope :by_categories, ->(category) { where(category: category) }
-  default_scope { order(created_at: :desc) }
+  scope :ordered, -> { order(created_at: :desc) }
 
-  scope :title_search, lambda { |title_search|
+  scope :search, lambda { |title_search|
```
Comment (okohuch) [inline app/models/news.rb]: follow the convention of `lambda` definition in the file

File: app/models/news.rb [hunk 2]
```diff
@@ -10,10 +10,11 @@ class News < ApplicationRecord
   validates :body, presence: true
   validates :short_information, presence: true, length: { maximum: 200 }
 
+  scope :by_views, -> { order(views: :desc) }
```
Comment (okohuch) [inline app/models/news.rb]: the name `by_views` is not very clear what it is
maybe `ordered_by_views`?

File: app/controllers/news_controller.rb [hunk 3]
```diff
@@ -2,14 +2,17 @@
 
 class NewsController < ApplicationController
   before_action :find_current_user_news, only: %i[edit update destroy]
+
+  has_scope :by_views, type: :boolean
```
Comment (okohuch) [inline app/controllers/news_controller.rb]: why do you need to have these here?

Comment (kmandziy) [reply app/controllers/news_controller.rb]: It adds query 'news?by_views=true'  or 'by_views=false' in request path

File: app/models/news.rb (hunk 2 above)
Comment (kmandziy) [reply app/models/news.rb]: changed that

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **96.19%** covered at **3.92** hits/line

File: app/models/news.rb (hunk 1 above)
Comment (kmandziy) [reply app/models/news.rb]: changed

File: app/controllers/news_controller.rb [hunk 4]
```diff
@@ -2,14 +2,17 @@
 
 class NewsController < ApplicationController
   before_action :find_current_user_news, only: %i[edit update destroy]
+
+  has_scope :ordered_by_views, type: :boolean
```
Comment (okohuch) [inline app/controllers/news_controller.rb]: do you need to apply these scopes to all controller actions or just to index?

File: app/views/news/index.html.haml [hunk 5]
```diff
@@ -3,32 +3,15 @@
...
-              = fa_icon 'tags', text: t('news.links.category'), class: 'ml-4'
-              = link_to news.category.name
-          .col-lg-4
-            .thumbnail
-              = image_tag news.image_url(:thumb)
+      %h2
+        = t('news.index.university_news')
+      %p
+        = t('news.index.sort_by')
+        = link_to t('news.links.popular'), '/news?ordered_by_views=true', remote: true,
```
Comment (okohuch) [inline app/views/news/index.html.haml]: you can change the path to `news_path(ordered_by_views: true)`

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **96.19%** covered at **3.92** hits/line

File: app/controllers/news_controller.rb (hunk 4 above)
Comment (kmandziy) [reply app/controllers/news_controller.rb]: Already added only index

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **96.19%** covered at **3.92** hits/line

**Pull request merged**